src/js/core/directives/ui-grid.js

```diff
@@ -84,12 +84,10 @@
         // gridUtil.logDebug('dataWatch fired');
         var promises = [];
 
-        if ( self.grid.options.fastWatch ){
```

Comment:
Removed the fastWatch condition for expecting data to be a string because based on line 62 of this file, strings for data are supported without fastWatch.